src/nrnoc/init.c

```diff
@@ -255,10 +258,6 @@ void hoc_nrn_load_dll(void) {
 	}else{
 		hoc_retpushx(0.);
 	}	
-#else
-	hoc_warning("nrn_load_dll not available on this machine", (char*)0);
-	hoc_retpushx(0.);
-#endif
```

Comment:
@nrnhines as you said the logic and relationships among different `ENABLE_NRN` flags is a bit fuzzy right now. However I suggest keeping these changes for `mswin_load_dll` since this feature should be available in any case.

Reply:
I believe this dates back to the time when some machines did not support  shared libraries.  Also prior to that the idea of dynamic loading after launch was first restricted to windows (hence the _dll in all the names of the functions) and only later did NRNMECH_DLL_STYLE become an option for linux via the linux_nrnmech autotools variable. As you can see from the removal of the ```#if defined(WIN32) || defined(NRNMECH_DLL_STYLE)``` the person who removed it considered it a no longer needed legacy. I think that is probably ok. And really, it could be removed in the other places it occurs in this file. The only other place the token substantively appears is in nrn/src/ivoc/ivocmain.cpp where it manages the -dll arg so that the name gets copied into nrn_mech_dll for later loading (when left as default "" the later loading is for the default "x86_64/.libs/libnrnmech.so").  So it seems that ultimately unset NRNMECH_DLL_STYLE could be made to mean that the binary special should link the mod files statically  (with the
consequence that one could not load them from python. Not a very important case.